### PR TITLE
Enable the quick dashboard when the toggle is turned on

### LINKED ISSUE #4450: [New Dashboard] Enabling of `quicker_dashboard_key` toggle do not trigger dashboard cache creation
##### Issue Type

- Bug Report

##### Summary

Enabling of `quicker_dashboard_key` toggle do not trigger dashboard cache creation. Empty dashboard page is shown to users. Need to perform a full config save to get it triggered and dashboard page to load with pipelines

###### Basic environment details

* Go Version: `18.2.0 (6175-1058e75b18e8a645dd71702851994a010789f450)`

##### Steps to Reproduce

1. On an existing GoCD setup with multiple pipeline and instances of it enable the toggle `quciker_dashboard_key`
2. Visit `<go-server>/go/dashbaord` page
3. Empty dashboard is shown up

##### Expected Results

Dashboard page should showup all the pipelines

##### Actual Results

Only on a full config save all pipelines are shown. Or if there are any pipeline runs, then only that specific pipeline is shown

### PR DESCRIPTION
**Pull request by ketan**
Hook in a listener in the `FeatureToggleService` which notifies
when the toggle is turned on, and pretends like a full config save happened

Fixes #4450

### CODE REVIEW

File: server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public void visit(PipelineConfigs group) {
     }
 
     public void updateCacheForPipeline(CaseInsensitiveString pipelineName) {
-        if (!Toggles.isToggleOn(Toggles.QUICKER_DASHBOARD_KEY)) {
+        if (featureToggleDisabled()) {
```
Comment (varshavaradarajan) [inline server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java]: rename this to`isFeatureToggleDisabled` ?

File: server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java [hunk 2]
```diff
@@ -75,14 +75,18 @@ public void updateCacheForPipeline(CaseInsensitiveString pipelineName) {
...
 
     public void updateCacheForPipeline(PipelineConfig pipelineConfig) {
-        if (!Toggles.isToggleOn(Toggles.QUICKER_DASHBOARD_KEY)) {
+        if (featureToggleDisabled()) {
             return;
         }
         updateCache(goConfigService.findGroupByPipeline(pipelineConfig.name()), pipelineConfig);
     }
 
+    public boolean featureToggleDisabled() {
```
Comment (varshavaradarajan) [inline server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java]: Does this belong here?

Comment (ketan) [reply server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java]: It's only temporary. I think.

File: server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java (hunk 1 above)
Comment (ketan) [reply server/src/main/java/com/thoughtworks/go/server/service/GoDashboardService.java]: Done. And merged.

**Pull request merged**